lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -5469,11 +5470,15 @@ void SwiftASTContextForExpressions::ModulesDidLoad(ModuleList &module_list) {
     std::vector<std::pair<std::string, bool>> framework_search_paths;
     std::vector<std::string> extra_clang_args;
     lldb::ModuleSP module_sp = module_list.GetModuleAtIndex(mi);
+    if (!module_sp)
```

Comment:
Should this be an assert instead? The module should never be null if iterating over the module list.

Reply:
A shared pointer to a [Module](https://lldb.llvm.org/cpp_reference/classlldb__private_1_1Module.html) which can contain NULL if idx is out of range.

Reply:
Since we are not using ForEach here, I'm not sure if other threads aren't allowed to modifiy the list while we're iterating.